docs/proposals/advanced-nginx-extensions.md

```diff
@@ -26,3 +74,915 @@ NGINX use cases. To allow them to implement those use cases, we need to bring a
+doesn't support filters, so it needs to be added to the Gateway API first.
+
+Although SnippetsFilter applicability is limited compared to SnippetsPolicy, SnippetsFilter creates
+a natural split of responsibilities between Cluster operator and Application developers:
+
+- Cluster operator creates a SnippetsFilter.
+- Application developers reference the SnippetsFilter to enable it.
+
+Note that with the SnippetsPolicy, because the targetRef is part of the SnippetsPolicy, it is not possible to have
+such a split of responsibilities.
+
+TO-DO(pleshakov) - Add SnippetsFilter to the GEP after discussions about it.
+
+### Personas
+
+The target persona of snippets is Cluster operator. They will create and manage snippets.
+
+Snippets are not intended for Application developers, because:
+
+- To create snippets, you need to know about NGINX and how NGF generates NGINX configuration.
+- Snippets can easily break NGINX config (if not validated by NGF). See
+  [Config Validation section](#nginx-config-validation) below.
+- Snippets can be used to exploit NGF. (See [Security Considerations section](#security-considerations) below)
+
+As mentioned in the [Policy or Filter](#policy-or-filter), when snippets are used via SnippetsFilter, Application
+developers can still control whether they want to enable or disable snippets by referencing them in an HTTPRoute.
+
+### NGINX Config Validation
+
+An invalid snippet can break NGINX config. When this happens, NGINX will continue to use the last valid configuration.
+However, any subsequent configuration updates (for example, caused by changes to an HTTPRoute) will not be possible
+until the
+invalid snippet is removed.
+
+Before injecting snippets in NGINX config, it is possible to validate snippets
+using [crossplane](https://github.com/nginxinc/nginx-go-crossplane),
+although some work is necessary to support it -- see https://github.com/nginxinc/nginx-go-crossplane/issues/94. Such
+validation will catch cases of using invalid directives or invalid parameters of directives. At the same time, it
+will not be a complete validation. For example, it won't catch errors like referencing an undefined variable
+in the NGINX config.
+
+### Security Considerations
+
+The snippet below exposes the filesystem of the NGINX container including any TLS Secrets and the ServiceAccount Secret,
+which NGF uses to access Kubernetes API:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsPolicy
+metadata:
+  name: expose
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: cafe
+  snippets:
+  - context: http.server
+    value: |
+      server {
+        listen 80;
+        root /;
+        autoindex on;
+      }
+```
+
+As a consequence, creating SnippetsPolicy should only be allowed for the privileged users -- Cluster operator.
+As a further precaution, we can disable SnippetsPolicy by default similarly
```

Comment:
I do think we should disable by default. Can we rephrase to make this a requirement?